test/unit/conditions/context.test.ts

```diff
@@ -77,6 +78,32 @@ describe('context parameters', () => {
     });
   });
 
+  it('detects if a signer is required', () => {
+    const conditionObj = {
+      ...testContractConditionObj,
+      returnValueTest: {
+        ...testReturnValueTest,
+        value: USER_ADDRESS_PARAM,
+      },
+    };
+    const condition = new ContractCondition(conditionObj);
+    const conditionExpr = new ConditionExpression(condition);
+    const conditionContext = conditionExpr.buildContext(provider, {}, signer);
+    expect(conditionExpr.contextRequiresSigner()).toBe(true);
+    expect(conditionContext.requiresSigner()).toBe(true);
+  });
+
+  it('detects if a signer is not required', () => {
+    const condition = new RpcCondition(testRpcConditionObj);
+    const conditionExpr = new ConditionExpression(condition);
+    const conditionContext = conditionExpr.buildContext(provider, {}, signer);
+    expect(JSON.stringify(condition.toObj()).includes(USER_ADDRESS_PARAM)).toBe(
+      false
+    );
+    expect(conditionExpr.contextRequiresSigner()).toBe(false);
+    expect(conditionContext.requiresSigner()).toBe(false);
```

Comment:
How come we need the check on both the `ConditionExpr` and the `Context` object? I would have assumed it's only needed in `ConditionExpr`.

Reply:
We don't necessarily need to check it - it's just when a wrapper for exposing this functionality is higher up in the API. I think it makes sense to add this check to the tests to document this API.

Reply:
I guess I really meant "expose" and not "check"

What's the case where the call `requiresSigner()` is performed on the `Context` instead of the `Condition`? Having it in the `Context` doesn't mesh with my understanding - the `Context` is just the dictionary of values, but it is the condition that determines whether a signer is needed or not.

Reply:
I think I had some use case in mind but on second thought it's already covered by `ContextCondition.validate`. I think we can make `ContextCondition.requiresSigner` private then.

Reply:
If still needed internally for validation - then yes, private would be better 👍 .